pkg/services/accesscontrol/resourcepermissions/api_test.go

```diff
@@ -0,0 +1,486 @@
+					Users:        true,
+					Teams:        false,
+					BuiltInRoles: false,
+				},
+				PermissionsToActions: map[string][]string{
+					"View": {"dashboards:read"},
+				},
+			},
+			permissions:    []*accesscontrol.Permission{},
+			expected:       Description{},
+			expectedStatus: http.StatusForbidden,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			_, server, _ := setupTestEnvironment(t, &models.SignedInUser{}, tt.permissions, tt.options)
+			req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("/api/access-control/%s/description", tt.options.Resource), nil)
+			require.NoError(t, err)
+			recorder := httptest.NewRecorder()
+			server.ServeHTTP(recorder, req)
+
+			got := Description{}
+			require.NoError(t, json.NewDecoder(recorder.Body).Decode(&got))
+			assert.Equal(t, tt.expected, got)
+			if tt.expectedStatus == http.StatusOK {
+				assert.Equal(t, tt.expectedStatus, recorder.Code)
+			}
+		})
+	}
+}
+
+type getPermissionsTestCase struct {
+	desc           string
+	resourceID     string
+	permissions    []*accesscontrol.Permission
+	expectedStatus int
+}
+
+func TestApi_getPermissions(t *testing.T) {
+	tests := []getPermissionsTestCase{
+		{
+			desc:           "expect permissions for resource with id 1",
+			resourceID:     "1",
+			permissions:    []*accesscontrol.Permission{{Action: "dashboards.permissions:read", Scope: "dashboards:id:1"}},
+			expectedStatus: 200,
+		},
+		{
+			desc:           "expect http status 403 when missing permission",
+			resourceID:     "1",
+			permissions:    []*accesscontrol.Permission{},
+			expectedStatus: 403,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			service, server, sql := setupTestEnvironment(t, &models.SignedInUser{OrgId: 1}, tt.permissions, testOptions)
+
+			// seed team 1 with "Edit" permission on dashboard 1
+			team, err := sql.CreateTeam("test", "test@test.com", 1)
+			require.NoError(t, err)
+			_, err = service.SetTeamPermission(context.Background(), team.OrgId, team.Id, tt.resourceID, []string{"dashboards:read", "dashboards:write", "dashboards:delete"})
+			require.NoError(t, err)
+			// seed user 1 with "View" permission on dashboard 1
+			u, err := sql.CreateUser(context.Background(), models.CreateUserCommand{Login: "test", OrgId: 1})
+			require.NoError(t, err)
+			_, err = service.SetUserPermission(context.Background(), u.OrgId, u.Id, tt.resourceID, []string{"dashboards:read"})
+			require.NoError(t, err)
+
+			// seed built in role Admin with "View" permission on dashboard 1
+			_, err = service.SetBuiltInRolePermission(context.Background(), 1, "Admin", tt.resourceID, []string{"dashboards:read", "dashboards:write", "dashboards:delete"})
+			require.NoError(t, err)
+
+			permissions, recorder := getPermission(t, server, testOptions.Resource, tt.resourceID)
+			assert.Equal(t, tt.expectedStatus, recorder.Code)
+
+			if tt.expectedStatus == http.StatusOK {
+				for _, p := range permissions {
```

Comment:
We could also check that we correctly retrieve the three expected permissions:
```suggestion
				assert.Len(t, permissions, 3, "expected three assignments: user, team, builtin")
				for _, p := range permissions {
```